orca-core/src/main/java/com/netflix/spinnaker/orca/TaskResolver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TaskResolver {
+  private final Map<String, Task> taskByAlias = new HashMap<>();
+
+  public TaskResolver(Collection<Task> tasks) {
+    for (Task task : tasks) {
+      taskByAlias.put(task.getClass().getCanonicalName(), task);
+      for (String alias: task.aliases()) {
+        taskByAlias.put(alias, task);
+      }
+    }
+  }
+
+  public Task getTask(String taskTypeIdentifier) {
```

Comment:
Thoughts on trying to enforce usage of `@Nonnull`, etc on args / returns within Java?

Reply:
I'm not in the habit of doing so but don't see the harm.